### ISSUE TITLE
Show a warning when building APK/AAB with unsupported architecture

### ISSUE THREAD

**Issue by xster** [flutter/flutter#61122]
See https://github.com/flutter/flutter/issues/37566 and https://github.com/flutter/flutter/issues/59501. 

It's possible for users to end up adding their own native libraries in their Flutter Android project's gradle. If the user ends up adding an x86 .so to their project, the APK will be installable on x86 physical devices, cause it to skip the ARM emulation and crash when trying to load Flutter since there are no flutter .so. 

In flutter.gradle, we should look for the presence of native libraries in the APK with unsupported architectures and show a warning to prevent this subtle crasher.

**Comment by chinmaygarde** [comment]
Agreed that this can cause subtle issues and a warning is necessary. Some other thoughts/ideas:
* Explicitly specify the architectures in Gradle files generated from the templates (Not sure how this will play with the architectures specified on the command line).
* Cause builds the fails by inspecting the APK post packaging to see if there are dylibs for architectures other than those supported by Flutter and/or specified on the command line.

**Comment by blasten** [comment]
I wonder if AGP has a hook to verify that. You might want to get input from the AGP team. I can see this causing race conditions, but there might be a way to work around those.
